evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -715,6 +715,7 @@ def test_automl_allowed_pipelines_specified_allowed_pipelines_binary(
         X_train=X,
         y_train=y,
         problem_type="binary",
+        optimize_thresholds=False,
```

Comment:
Following up on [my outdated comment](https://github.com/alteryx/evalml/pull/2320#discussion_r646970921) on another one of these similar lines:
> Why set these to False?
> 
> If you did it to avoid having the tests waste time calling the threshold optimizer, then, lol I agree we shouldn't run the optimizer in every test. I do wonder if there's a different way to accomplish this though. Could we mock `BinaryClassificationObjective.optimize_threshold` instead, in the same way we mock pipeline fit and score in many tests?

I wasn't just referring to that specific test though heh, I was referring to every automl test where you've added `optimize_thresholds=False` in this PR.

Reply:
Ah, I added `optimize_thresholds` to false to avoid tuning the thresholds. The other way would've been to patch `predict_proba`, `optimize_thresholds`, and `encode_targets` since so many of the tests patch the pipeline `fit`.

Reply:
Oh damn yeah I follow. Basically, now that we default to enabling threshold optimization, even if we mock pipeline `fit`/`predict`/`score`, optimization will still run and consume a bunch of test runtime.

@freddyaboulton FYI this could be relevant for #1815 #2298 

Thanks for explaining @bchen1116 SGTM